test/jb_eth_payment_terminal/redeem_tokens_of.test.js

```diff
@@ -0,0 +1,266 @@
+    let jbOperations = await jbOperationsFactory.deploy();
+    const REDEEM_PERMISSION_INDEX = await jbOperations.REDEEM();
+
+    /* Common mocks */
+
+    await mockJbOperatorStore.mock.hasPermission
+      .withArgs(holder.address, holder.address, PROJECT_ID, REDEEM_PERMISSION_INDEX)
+      .returns(true);
+
+    const fundingCycle = {
+      number: FUNDING_CYCLE_NUM,
+      configuration: timestamp,
+      basedOn: timestamp,
+      start: timestamp,
+      duration: 0,
+      weight: WEIGHT,
+      discountRate: 0,
+      ballot: ethers.constants.AddressZero,
+      metadata: 0,
+    };
+
+    return {
+      beneficiary,
+      holder,
+      jbEthPaymentTerminal,
+      fundingCycle,
+      mockJbEthPaymentTerminalStore,
+      mockJbOperatorStore,
+      otherCaller,
+      timestamp,
+    };
+  }
+
+  it('Should redeem tokens for overflow and emit event', async function () {
+    const {
+      beneficiary,
+      fundingCycle,
+      holder,
+      jbEthPaymentTerminal,
+      mockJbEthPaymentTerminalStore,
+      timestamp,
+    } = await setup();
+
+    // Keep it simple and let 1 token exchange for 1 wei
+    await mockJbEthPaymentTerminalStore.mock.recordRedemptionFor
+      .withArgs(
+        holder.address,
+        PROJECT_ID,
+        /* tokenCount */ AMOUNT,
+        /* minReturnedWei */ AMOUNT,
+        beneficiary.address,
+        MEMO,
+        /* delegateMetadata */ 0,
+      )
+      .returns(fundingCycle, /* claimAmount */ AMOUNT, MEMO);
+
+    // Give terminal sufficient ETH
+    await ethers.provider.send('hardhat_setBalance', [
+      jbEthPaymentTerminal.address,
+      '0x' + AMOUNT.toString(16),
+    ]);
+
+    const initialBeneficiaryBalance = await ethers.provider.getBalance(beneficiary.address);
+
+    const tx = await jbEthPaymentTerminal
+      .connect(holder)
+      .redeemTokensOf(
+        holder.address,
+        PROJECT_ID,
+        /* tokenCount */ AMOUNT,
+        /* minReturnedWei */ AMOUNT,
+        beneficiary.address,
+        MEMO,
+        /* delegateMetadata */ 0,
+      );
+
+    expect(await tx)
+      .to.emit(jbEthPaymentTerminal, 'RedeemTokens')
+      .withArgs(
+        /* _fundingCycle.configuration */ timestamp,
+        /* _fundingCycle.number */ FUNDING_CYCLE_NUM,
+        /* _projectId */ PROJECT_ID,
+        /* _holder */ holder.address,
+        /* _beneficiary */ beneficiary.address,
+        /* _tokenCount */ AMOUNT,
+        /* claimAmount */ AMOUNT,
+        /* memo */ MEMO,
+        /* msg.sender */ holder.address,
+      );
+
+    // Terminal should be out of ETH
+    expect(await ethers.provider.getBalance(jbEthPaymentTerminal.address)).to.equal(0);
+
+    // Beneficiary should have a larger balance
+    expect(await ethers.provider.getBalance(beneficiary.address)).to.equal(
+      initialBeneficiaryBalance.add(AMOUNT),
+    );
+  });
+
+  it('Should only emit event if claim amount is 0', async function () {
```

Comment:
confused about the description string here. What event should only be emitted is claim amount is 0?

Reply:
the event should still be emitted, but no amount transferred to the beneficiary. The test looks right, the string off.